felix/dataplane/linux/int_dataplane.go

```diff
@@ -683,23 +685,43 @@ func NewIntDataplaneDriver(config Config) *InternalDataplane {
 
 	// If no overlay is enabled, and Felix is responsible for programming routes, starts a manager to
 	// program no encapsulation routes.
-	if config.ProgramClusterRoutes &&
-		!config.RulesConfig.VXLANEnabled && !config.RulesConfig.IPIPEnabled && !config.RulesConfig.WireguardEnabled {
-		log.Info("No encapsulation enabled, starting thread to keep no encapsulation routes in sync.")
-		// Add a manager to keep the all-hosts IP set up to date.
-		dp.noEncapManager = newNoEncapManager(
-			routeTableV4,
-			4,
-			config,
-			dp.loopSummarizer,
-		)
-		dp.noEncapParentIfaceC = make(chan string, 1)
-		go dp.noEncapManager.monitorParentDevice(
-			context.Background(),
-			time.Second*10,
-			dp.noEncapParentIfaceC,
-		)
-		dp.RegisterManager(dp.noEncapManager)
+	if config.ProgramClusterRoutes {
+		if !config.RulesConfig.VXLANEnabled && !config.RulesConfig.IPIPEnabled && !config.RulesConfig.WireguardEnabled {
+			log.Info("No encapsulation enabled, starting thread to keep no encapsulation routes in sync.")
+			// Add a manager to keep the all-hosts IP set up to date.
+			dp.noEncapManager = newNoEncapManager(
+				routeTableV4,
+				4,
+				config,
+				dp.loopSummarizer,
+			)
+			dp.noEncapParentIfaceC = make(chan string, 1)
+			go dp.noEncapManager.monitorParentDevice(
+				context.Background(),
+				time.Second*10,
+				dp.noEncapParentIfaceC,
+			)
+			dp.RegisterManager(dp.noEncapManager)
+		}
+
+		if config.IPv6Enabled &&
+			!config.RulesConfig.VXLANEnabledV6 && !config.RulesConfig.WireguardEnabledV6 {
+			log.Info("No IPv6 encapsulation enabled, starting thread to keep no encapsulation routes in sync.")
```

Comment:
```suggestion
			log.Info("Unencapsulated IPv6 route programming enabled, starting thread to keep no encapsulation routes in sync.")
```

"No IPv6 encapsulation enabled" is subtly different than "Unencapsualated IPv6 routing enabled"